db: add new GetterRun.latest() method

R2ZER0 opened the pull request:
Add a new GetterRun.latest() class method to easily fetch the most recent GetterRun object, and replace the previous uses of objects.last() which probably does return the most recent but theoretically isn't guaranteed to.

https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.latest
https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.last

Questions:
* Maybe we should call the method something a bit more descriptive like `latest_run()` or `latest_run_object()` instead of just `latest()`?
* Black-format-on-save strikes again, should I reinstate the removed whitespace or are we aiming to gradually reformat everything?

michaelwood commented:
> 
> Questions:
> 
> * Maybe we should call the method something a bit more descriptive like `latest_run()` or `latest_run_object()` instead of just `latest()`?

I think in this case because we have the function name-spaced by `GetterRun` this descriptive enough. No objections if you come up with a better function name though.

> * Black-format-on-save strikes again, should I reinstate the removed whitespace or are we aiming to gradually reformat everything?

Everything in this repo is python black formatted but you may have a different version that is being run by a plugin rather than provided by the virtualenv? 

https://github.com/ThreeSixtyGiving/datastore/blob/master/requirements_dev.txt#L27
https://github.com/ThreeSixtyGiving/datastore/blob/master/.github/workflows/lint.yml#L24

python black is deliberately not pinned so that we have a rolling programme of keeping the code formatted with latest so we may need to update python black soon. Either way happy to have linting stuff but ideally in a separate commit as much as possible.

R2ZER0 committed:
db: add new GetterRun.latest() method

Add a new GetterRun.latest() class method to easily fetch the most
recent GetterRun object, and replace the previous uses of objects.last()
which probably does return the most recent but theoretically isn't
guaranteed to.

R2ZER0 commented:
> I think in this case because we have the function name-spaced by `GetterRun` this descriptive enough. No objections if you come up with a better function name though.

Happy to keep it as `.latest()` then!

> Everything in this repo is python black formatted but you may have a different version that is being run by a plugin rather than provided by the virtualenv?

Aha, that is exactly what was happening.

> https://github.com/ThreeSixtyGiving/datastore/blob/master/requirements_dev.txt#L27 https://github.com/ThreeSixtyGiving/datastore/blob/master/.github/workflows/lint.yml#L24
> 
> python black is deliberately not pinned so that we have a rolling programme of keeping the code formatted with latest so we may need to update python black soon. Either way happy to have linting stuff but ideally in a separate commit as much as possible.

I re-ran against the version of black that's in `dev_requirements.txt` and it didn't make the same whitespace changes. The VSCode extension I was using (Trunk) had black 23 vs the venv's 22.

@michaelwood I guess I could create another PR to upgrade `dev_requirements.txt` to 23 and reformat everything... do we want to do that?

For this PR, Ive excluded the black 22->23 changes.

michaelwood commented:
> 
> @michaelwood I guess I could create another PR to upgrade `dev_requirements.txt` to 23 and reformat everything... do we want to do that?
> 

I don't think we need to start doing that work now. The next time someone does an upgrade of the python packages it can be grouped with that work.
